### PR TITLE
chore: use same server info for all map modes and refactor

### PR DESCRIPTION
**Pull request by kohlisid**
Fixes : [#1837](https://github.com/numaproj/numaflow/issues/1837)

### CODE REVIEW

**Comment by kohlisid** [comment]
Map Stream

```
 skohli@macos-JQWR9T560R flatmap_stream % kubectl logs simple-pipeline-flatmap-0-nzluc -f  
+ python example.py
2024-07-16 21:03:01 INFO     Starting Map Stream Server
2024-07-16 21:03:01 INFO     Async GRPC Server listening on: unix:///var/run/numaflow/mapstream.sock with max threads: 4

```

```
 kubectl logs simple-pipeline-flatmap-0-nzluc numa -f  | grep "mapper"
2024/07/16 21:02:43 Server info file /var/run/numaflow/mapper-server-info is not ready...

 kubectl logs simple-pipeline-flatmap-0-nzluc numa -f  | grep Info
2024/07/16 21:03:01 ServerInfo: &{uds python 1.2.0-rc4 0.7.3 map[MAP_MODE:stream-map]}

kubectl logs simple-pipeline-flatmap-0-nzluc numa -f  | grep "enabled"
2024-07-16T21:03:01.93051948Z   INFO    numaflow.MapUDF-processor       udf/map_udf.go:150      Map mode enabled: Stream Map    {"pipeline": "simple-pipeline", "vertex": "flatmap", "protocol": "uds-grpc-map-udf"}



```

File: pynumaflow/info/types.py
Comment (KeranYang) [inline pynumaflow/info/types.py:28]: nit:
```suggestion
# MAP_MODE_KEY is used in the server info object's metadata to indicate which map mode is enabled
MAP_MODE_KEY = "MAP_MODE"
```

**Comment by kohlisid** [comment]
Batch Map with new map mode metadata

no annotations in spec
```
    - name: batch-flatmap
      partitions: 2
      scale:
        min: 1
      udf:
        container:
          image: quay.io/kohlisid/numaflow-python/batch-map-flatmap:stable1
          imagePullPolicy: Always
```
```
2024/07/17 21:06:52 Server info file /var/run/numaflow/mapper-server-info is not ready...
2024/07/17 21:06:53 ServerInfo: &{uds python 1.2.0-rc4 0.7.3 map[MAP_MODE:batch-map]}
{"level":"info","ts":"2024-07-17T21:06:53.813018858Z","logger":"numaflow.MapUDF-processor","caller":"udf/map_udf.go:173","msg":"Map mode enabled: Batch Map","pipeline":"flatmap","vertex":"batch-flatmap","protocol":"uds-grpc-map-udf"}
2024/07/17 21:06:53 UDS Client: unix:/var/run/numaflow/batchmap.sock

```

**Comment by kohlisid** [comment]
Unary Map with new map mode metadata

no annotations in spec
```
    - name: batch-flatmap
      partitions: 2
      scale:
        min: 1
      udf:
        container:
          image: quay.io/kohlisid/numaflow-python/batch-map-flatmap:stable1
          imagePullPolicy: Always
```


```

2024/07/17 21:13:39 Server info file /var/run/numaflow/mapper-server-info is not ready..
2024-07-17T21:13:55.67101085Z   INFO    numaflow.MapUDF-processor       udf/map_udf.go:196      Map mode enabled: Unary Map     {"pipeline": "simple-pipeline", "vertex": "flatmap", "protocol": "uds-grpc-map-udf"} 
2024/07/17 21:13:55 ServerInfo: &{uds python 1.2.0-rc4 0.7.3 map[MAP_MODE:unary-map]}
2024/07/17 21:13:55 UDS Client: unix:/var/run/numaflow/map.sock

```

**Comment by kohlisid** [comment]
Unary Map with no metadata (older code + new numaflow backward)

no annotations in spec
```
    - name: flatmap
      udf:
        container:
          image: quay.io/kohlisid/numaflow-python/map-flatmap:stable
          imagePullPolicy: Always
          env:
            - name: PYTHONDEBUG
              value: "true"
```


```

2024/07/17 21:21:21 Server info file /var/run/numaflow/mapper-server-info is not ready...
2024/07/17 21:21:34 ServerInfo: &{uds python 1.2.0-rc4 0.7.3 map[]}
2024-07-17T21:21:34.368700528Z  INFO    numaflow.MapUDF-processor       udf/map_udf.go:196      Map mode enabled: Unary Map     {"pipeline": "simple-pipeline", "vertex": "flatmap", "protocol": "uds-grpc-map-udf"}
2024/07/17 21:21:34 UDS Client: unix:/var/run/numaflow/map.sock


```

**Comment by kohlisid** [comment]
Multiproc Map
```
2024-07-17T21:27:54.898644917Z  INFO    numaflow.MapUDF-processor       udf/map_udf.go:196      Map mode enabled: Unary Map     {"pipeline": "simple-pipeline", "vertex": "mult", "protocol": "uds-grpc-map-udf"}
2024/07/17 21:27:54 ServerInfo: &{uds python 1.2.0-rc4 0.7.3 map[CPU_LIMIT:6 MULTIPROC:3]}
2024/07/17 21:27:54 Multiprocessing Servers : 3



```

**Review by vigith — APPROVED** [review]

**Review by ayildirim21 — APPROVED** [review]

**Review by KeranYang — APPROVED** [review]

**Comment by KeranYang** [comment]
I am approving but it would be great if we make the codecov pass before merging.

**Pull request merged**
